src/mlpack/tests/one_hot_encoding_test.cpp

```diff
@@ -0,0 +1,231 @@
+/**
+ * @file tests/one_hot_encoding_test.cpp
+ * @author Jeffin Sam
+ *
+ * Tests for the One-Hot Encoding method.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include "test_catch_tools.hpp"
+#include "catch.hpp"
+#include <mlpack/core/data/one_hot_encoding.hpp>
+
+using namespace mlpack;
+using namespace mlpack::data;
+using namespace std;
+
+/**
+ * Test one hot encoding for small input.
+ */
+TEST_CASE("OneHotEncodingTest01", "[OneHotEncodingTest]")
+{
+  arma::Mat<size_t> matrix;
+  matrix = "1 0;"
+           "0 1;"
+           "1 0;"
+           "1 0;"
+           "1 0;"
+           "1 0;"
+           "0 1;"
+           "1 0;";
+  // Output matrix to save one-hot encoding results.
+  arma::Mat<size_t> output;
+  matrix = matrix.t();
+  arma::irowvec labels("-1 1 -1 -1 -1 -1 1 -1");
+  data::OneHotEncoding(labels, output);
+
+  REQUIRE(matrix.n_cols == output.n_cols);
+  REQUIRE(matrix.n_rows == output.n_rows);
+  CheckMatrices(output, matrix);
+}
+
+/**
+ * Test one hot encoding for sparse matrix.
+ */
+TEST_CASE("OneHotEncodingSparseMatTest", "[OneHotEncodingTest]")
+{
+  arma::SpMat<size_t> matrix;
+  matrix = "1 0;"
+           "0 1;"
+           "1 0;"
+           "1 0;"
+           "1 0;"
+           "1 0;"
+           "0 1;"
+           "1 0;";
+
+  matrix = matrix.t();
+  // Output matrix to save onehotencoding results.
```

Comment:
```suggestion
  // Output matrix to save one-hot encoding  results.
```

Just to be a little bit more consistent with the naming.